app/clients/__init__.py

```diff
@@ -13,6 +13,7 @@
         "addressing_style": "virtual",
     },
     use_fips_endpoint=True,
+    max_pool_connections=50,
```

Comment:
What is this config setting for?  Is it related to the `ThreadPoolExecutor` workers so we don't overwhelm S3 or make too many API calls at once?

Reply:
I think it's just to scale up/down how concurrent we want to be.  It's hard to overwhelm S3 (assuming the partition issue is fixed).   I didn't see any performance improvement on my local machine changing from 10 connections to 50, but just wanted to document that this is a lever we can pull.  I'm not sure why you are seeing the 50, I thought I reverted back to 10.  Maybe I didn't push.  It should be 10 for now.